sdk/src/client/node_manager/syncing.rs

```diff
@@ -66,63 +68,103 @@ impl ClientInner {
+                    Err(err) => {
+                        log::error!("Couldn't get node info: {err}");
                     }
                 }
-                Err(err) => {
-                    log::error!("Couldn't get node info: {err}");
+            } else {
+                match crate::client::Client::get_node_info(node.url.as_ref(), node.auth.clone()).await {
+                    Ok(info) => {
+                        if info.status.is_healthy || ignore_node_health {
+                            // Unwrap: We should always have parameters for this version. If we don't we can't recover.
+                            let protocol_parameters = info
+                                .protocol_parameters_by_version(PROTOCOL_VERSION)
+                                .expect("missing v3 protocol parameters")
+                                .parameters
+                                .clone();
+                            let network_name = protocol_parameters.network_name();
+                            match network_nodes.get_mut(network_name) {
+                                Some(network_node_entry) => {
+                                    network_node_entry.push((
+                                        protocol_parameters,
+                                        node.clone(),
+                                        info.status.relative_accepted_tangle_time,
+                                    ));
+                                }
+                                None => {
+                                    network_nodes.insert(
+                                        network_name.to_owned(),
+                                        vec![(
+                                            protocol_parameters,
+                                            node.clone(),
+                                            info.status.relative_accepted_tangle_time,
+                                        )],
+                                    );
+                                }
+                            }
+                        } else {
+                            log::warn!("{} is not healthy: {:?}", node.url, info);
+                        }
+                    }
+                    Err(err) => {
+                        log::error!("Couldn't get node info: {err}");
+                    }
                 }
             }
         }
 
-        // Get network_id with the most nodes
-        let mut most_nodes = ("network_id", 0);
-        for (network_id, node) in &network_nodes {
-            if node.len() > most_nodes.1 {
-                most_nodes.0 = network_id;
-                most_nodes.1 = node.len();
+        // Get the network_name that most nodes have in common
+        let mut most_nodes = ("network_name", 0);
+        for (network_name, nodes) in &network_nodes {
+            if nodes.len() > most_nodes.1 {
+                most_nodes.0 = network_name;
+                most_nodes.1 = nodes.len();
```

Comment:
this can be simplified into a one-liner by using `max_by_key`

Reply:
Good call, can actually simplify even more with the part below

Reply:
yeah, looks good